src/fondant/compiler.py

```diff
@@ -302,33 +291,47 @@ def kfp_pipeline():
                     k: v for k, v in component_op.arguments.items() if v is not None
                 }
 
-                # # Set image pull policy to always
-                # Execute the Kubeflow component and pass in the output manifest path from
-                # the previous component.
+                metadata = Metadata(
+                    pipeline_name=pipeline.name,
+                    run_id=run_id,
+                    base_path=pipeline.base_path,
+                    component_id=component_name,
+                    cache_key=component_op.get_component_cache_key(),
+                )
 
-                if previous_component_task is not None:
-                    component_task = kubeflow_component_op(
-                        input_manifest_path=manifest_path,
-                        metadata=metadata.to_json(),
-                        **component_args,
-                    )
-                    component_task.after(previous_component_task)
+                output_manifest_path = (
+                    f"{pipeline.base_path}/{pipeline.name}/"
+                    f"{run_id}/{component_name}/manifest.json"
+                )
+                # Set the execution order of the component task to be after the previous
+                # component task.
+                if component["dependencies"]:
+                    for dependency in component["dependencies"]:
```

Comment:
This won't work for more then one dependency, right? Maybe we should make this explicit by checking that the amount of dependencies is 1 and selecting the first one.